src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -30,27 +30,16 @@ export default class MongooseAdapter {
    * by id and should return all documents.
    */
   find(type, idOrIds, fields, sorts, filters, includePaths) {
-    let model = this.getModel(this.constructor.getModelName(type));
+    const model = this.getModel(this.constructor.getModelName(type));
     let queryBuilder = new mongoose.Query(null, null, model, model.collection);
-    let pluralizer = this.inflector.plural;
-    let mode = "find", idQuery;
-    let primaryDocumentsPromise, includedResourcesPromise = Q(null);
-
-    if(idOrIds) {
-      if(typeof idOrIds === "string") {
-        mode = "findOne";
-        idQuery = idOrIds;
-      }
-      else {
-        idQuery = {"$in": idOrIds};
-      }
+    const pluralizer = this.inflector.plural;
+    let primaryDocumentsPromise;
+    let includedResourcesPromise = Q(null);
 
-      queryBuilder[mode]({"_id": idQuery});
-    }
+    let idQuery = this.constructor.makeIdQuery(idOrIds);
+    let mode = typeof idQuery === 'string' ? 'findOne' : 'find';
 
-    else {
-      queryBuilder.find();
-    }
+    queryBuilder[mode](idQuery);
```

Comment:
Note that this isn't quite logically equivalent to the prior version. Before, if `idOrIds` was not truthy (as it is when a whole collection is looked up), `queryBuilder.find()` was called with no arguments; now, `queryBuilder.find()` will be called with `undefined` (the value of `idQuery` in that case). Since Mongoose often switches behavior based on the argument count, this may do something different; you'll want to test.